File: Java/curso_programacao/src/Diagonal_negativos.java
import java.util.Scanner;

public class Diagonal_negativos {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n,i,j, qt;
		qt=0;
		
		System.out.print("Qual a ordem da matriz: ");
		 n = sc.nextInt();
		 
		 int[][] mat = new int[n][n];
		 
		 for(i=0;i<n;i++) {
			for(j=0;j<n;j++) {
				System.out.print("Elemento ["+i+","+j+"]: ");
				mat[i][j] = sc.nextInt();
				if (mat[i][j]<0) {
				 qt=qt+1;
				}
			}
		 }
		 
		 System.out.println("Diagonal principal: ");
		 for(i=0;i<n;i++) {
			 System.out.print(mat[i][i] + " ");
		 }
		 System.out.println();
		 System.out.println("Quantidade de numeros negativos: " + qt);
		 
	}

}

File: Python/Diagonal negativos.py
n = int(input("Qual a ordem da matriz: "))

mat = [[0 for x in range (n)]for x in range (n)]
soma=0
for i in range(0, n):
   for j in range(0,n):
       mat[i][j] = int(input(f"Elemento [{i},{j}]: "))
       if mat[i][j] < 0:
           soma=soma+1

print("Diagonal principal: ")
for i in range(0,n):
    print(f"{mat[i][i]} ", end="")

print()
print(f"Numeros negativos: {soma}")